scheduler/src/cook/mesos/util.clj

```diff
@@ -701,6 +701,17 @@
                                 batch))))
     uncommitted-before))
 
+(defn clear-old-uncommitted-jobs-fn
+  "Returns a function that will, when invoked, clear any uncommitted jobs older than a week if the leadership atom is true."
+  [conn mesos-leadership-atom]
+  (fn [_]
```

Comment:
Instead of returning a function here, I think it makes more sense to have this perform the cleanup directly, and have the chime call `(fn [] (clear-old-uncommitted-jobs conn mesos-leadership-atom)`